src/client.ts

```diff
@@ -355,6 +390,69 @@ export class Client {
     }
   }
 
+  /**
+   * This Method allows the user to take a screenshot. If no selector is passed
+   * then a full page screenshot is saved.
+   * if no filename is passed, then file is stored with a timestamp as the filename
+   * @param params filename, selector
+   * @returns Filename
+   */
+  public async takeScreenshot(
+    params?: { fileName?: string; selector?: string },
```

Comment:
I think params should be required, a user should need to pass in a filename at the very least

Reply:
but doesn't an auto generated filename take less think to think about?

Reply:
if the user wants an auto generated filename, they should be responsible for that, i dont think that responsibility falls into the hands of this client method

And who's to say that users want a timestamped filename? i'd say it's very opinionated